bundles-lib/src/main/java/org/apache/metron/bundles/BundleClassLoaders.java

```diff
@@ -0,0 +1,353 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.bundles;
+
+import java.net.URISyntaxException;
+import java.util.*;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.commons.vfs2.FileSystemManager;
+import org.apache.metron.bundles.bundle.Bundle;
+import org.apache.metron.bundles.bundle.BundleCoordinate;
+import org.apache.metron.bundles.bundle.BundleDetails;
+import org.apache.metron.bundles.util.BundleProperties;
+import org.apache.metron.bundles.util.BundleSelector;
+import org.apache.metron.bundles.util.FileUtils;
+import org.apache.metron.bundles.util.BundleUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A singleton class used to initialize the extension and framework classloaders.
+ */
+public final class BundleClassLoaders {
+
+  private static volatile BundleClassLoaders bundleClassLoaders;
+  private volatile InitContext initContext;
+  private static final Logger logger = LoggerFactory.getLogger(BundleClassLoaders.class);
+
+  private final static class InitContext {
+
+    private final List<FileObject> extensionDirs;
+    private final Map<String, Bundle> bundles;
+
+    private InitContext(
+        final List<FileObject> extensionDirs,
+        final Map<String, Bundle> bundles) {
+      this.extensionDirs = extensionDirs;
+      this.bundles = bundles;
+    }
+  }
+
+  private BundleClassLoaders() {
+  }
+
+  /**
+   * @return The singleton instance of the BundleClassLoaders
+   */
+  public static BundleClassLoaders getInstance() {
+    BundleClassLoaders result = bundleClassLoaders;
+    if (result == null) {
+      synchronized (BundleClassLoaders.class) {
+        result = bundleClassLoaders;
+        if (result == null) {
+          bundleClassLoaders = result = new BundleClassLoaders();
+        }
+      }
+    }
+    return result;
+  }
+
+  public static void reset() {
+    getInstance().unInit();
+  }
+
+  private void unInit() {
+    synchronized (this) {
+      initContext = null;
+    }
+  }
```

Comment:
The reset() method appears to only be used for testing, is that correct?  It seems like the sort of thing that might be grossly misused by another developer who didn't understand that.
- Can it be less than "public"?
- Can it be annotated with the guava @VisibleForTesting annotation?
- Otherwise (or also) please javadoc reset() as not intended for use, and comment unInit() the same.